pkg/topology/run.go

```diff
@@ -406,3 +407,54 @@ func Run(opts TopologyRunOptions, s v1.SystemTemplate) []*pkg.Component {
 	}
 	return results
 }
+
+func SyncComponents(opts TopologyRunOptions, systemTemplate v1.SystemTemplate) error {
+	components := Run(opts, systemTemplate)
+	systemTemplateID, err := uuid.Parse(systemTemplate.GetPersistedID())
+	if err != nil {
+		return fmt.Errorf("failed to parse system template id: %w", err)
+	}
+
+	var compIDs []uuid.UUID
+	for _, component := range components {
+		component.Name = systemTemplate.Name
+		component.Namespace = systemTemplate.Namespace
+		component.Labels = systemTemplate.Labels
+		component.SystemTemplateID = &systemTemplateID
+		componentsIDs, err := db.PersistComponent(component)
+		if err != nil {
+			return fmt.Errorf("failed to persist component(id=%s, name=%s): %w", component.ID, component.Name, err)
+		}
+
+		compIDs = append(compIDs, componentsIDs...)
+	}
+
+	dbCompsIDs, err := db.GetActiveComponentsIDsWithSystemTemplateID(systemTemplateID.String())
+	if err != nil {
+		return fmt.Errorf("failed to get components: %w", err)
+	}
+
+	deleteCompIDs := difference(dbCompsIDs, compIDs)
+	if deleteCompIDs != nil {
+		if err := db.DeleteComponentsWithIDs(deleteCompIDs, time.Now()); err != nil {
+			return fmt.Errorf("failed to delete components: %w", err)
+		}
+	}
+
+	return nil
+}
+
+// difference returns the elements in `a` that aren't in `b`.
+func difference(a, b []uuid.UUID) []string {
```

Comment:
We already have a utils.SetDifference for this

Can you convert it into a generic function which takes `T comparable` ?